client/my-sites/upgrades/domain-management/dns/dns-list.jsx

```diff
@@ -43,7 +43,11 @@ const DnsList = React.createClass( {
 	addDns: function( record ) {
 		addDnsAction( this.props.selectedDomainName, record, ( error ) => {
 			if ( error ) {
-				notices.error( error.message );
+				notices.error( error.message || this.translate( 'The DNS record could not be restored.' ) );
+			} else {
+				this.props.successNotice( this.translate( 'The DNS record has been restored.' ), {
+					duration: 5000
```

Comment:
Well, you both make an argument that is easier for developers to use common notice bus and that is true :)
I am trying to figure out what makes more sense for users.
So for example if I make a mistake in an `IP address` for an `A` record, the most helpful information would be near the field I made a mistake in, not at the top of the page.

That way I dont even have to read it to know that I made a mistake.

> They require additional state keeping,

Yes, that is true.

> occupies space in the render tree 

Global notices render the same `<Notice>` component at the top of the page. It is precisely the same amount of components :)

Global notices ARE easy to use. But, because they dont know where did error come from, they are loosing visual information which could help users quickly see where the errors are.

Im just pointing room for improvement. It probably is an issue for a different PR, but maybe something to discuss or think about.

> PS: no, you go to sleep!

I have a superpower! I'm keeping US working hours to avoid Jetlag in between overseas trips :P